src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateManaged/InputBuffer.cs

```diff
@@ -72,26 +72,28 @@ public bool EnsureBitsAvailable(int count)
         /// </summary>
         public uint TryLoad16Bits()
         {
-            Debug.Assert(_buffer != null);
             if (_bitsInBuffer < 8)
             {
-                if (_start < _end)
+                if (!_buffer.IsEmpty)
                 {
-                    _bitBuffer |= (uint)_buffer[_start++] << _bitsInBuffer;
+                    _bitBuffer |= (uint)_buffer.Span[0] << _bitsInBuffer;
+                    _buffer = _buffer.Slice(1);
                     _bitsInBuffer += 8;
                 }
 
-                if (_start < _end)
+                if (!_buffer.IsEmpty)
                 {
-                    _bitBuffer |= (uint)_buffer[_start++] << _bitsInBuffer;
+                    _bitBuffer |= (uint)_buffer.Span[0] << _bitsInBuffer;
+                    _buffer = _buffer.Slice(1);
                     _bitsInBuffer += 8;
                 }
```

Comment:
This IsEmpty if block can be moved into the one above it. Also, to avoid doing multiple slices and in particular accessing .Span multiple times, I think it'd be better structured like:
```C#
if (_buffer.Length > 1)
{
    Span<byte> span = _buffer.Span;
    _bitBuffer |= (uint)span[0] << _bitsInBuffer;
    _bitBuffer |= (uint)span[1] << (_bitsInBuffer + 8);
    _bitsInBuffer += 16;
    _buffer = _buffer.Slice(2);
}
else if (_buffer.Length != 0)
{
    _bitBuffer |= (uint)_buffer.Span[0] << _bitsInBuffer;
    _buffer = _buffer.Slice(1);
    _bitsInBuffer += 8;
}
```

Reply:
@stephentoub Is there a cost associated with accessing the `Span` property on a memory instance? Good to know.

Reply:
Yes.  It's not huge, but it's not free.

Reply:
Fixed.